src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -117,38 +159,11 @@ export const _buildWaypointModelListFromRoute = (spawnPatternJson, navigationLib
  * @param airport
  * @return {array<object>}
  */
-export const _preSpawn = (spawnPatternJson, navigationLibrary, airport) => {
-    // find last fix along STAR that is outside of airspace, ie: next fix is within airspace
-    // distance between closest fix outside airspace and airspace border in nm
-    let extra = 0;
-    let totalDistance = 0;
+const _preSpawn = (spawnPatternJson, navigationLibrary, airport) => {
     // distance between each arriving aircraft, in nm
     const entrailDistance = spawnPatternJson.speed / spawnPatternJson.rate;
     const waypointModelList = _buildWaypointModelListFromRoute(spawnPatternJson, navigationLibrary, airport);
-
-
-    for (let i = 0; i < waypointModelList.length; i++) {
-        const waypoint = waypointModelList[i];
-        const waypointPosition = waypoint.position;
-        let previousWaypoint = waypoint;
-        let previousPosition = waypoint.position;
-
-        if (i > 0) {
-            previousWaypoint = waypointModelList[i - 1];
-            previousPosition = previousWaypoint.position;
-        }
-
-        if (isWithinAirspace(airport, waypointPosition) && i > 0) {
-            extra = nm(calculateDistanceToBoundary(airport, previousPosition));
-
-            continue;
-        }
-
-        // this will only work for `StandardRouteWaypointModel` objects. _buildWaypointModelListFromRoute may also return
-        // `FixModels`, in which case this line will return `NaN`
-        totalDistance += waypoint.distanceFromPreviousWaypoint;
-    }
-
+    const { totalDistance, distanceFromClosestFixToAirspaceBoundary } = _calculateDistancesAlongRoute(waypointModelList, airport);
```

Comment:
this was a really good suggestion.  it makes this thing sooooo much easier to read!